zkvm/src/blockchain/state.rs

```diff
@@ -60,4 +60,29 @@ impl BlockchainState {
 
         Ok(())
     }
+
+    /// Executes a list of transactions, returning their tx IDs and tx logs.
+    pub fn execute_txlist(
+        txs: Vec<Tx>,
+        version: u64,
+        timestamp_ms: u64,
+    ) -> Result<Vec<(TxID, TxLog)>, BlockchainError> {
+        let bp_gens = BulletproofGens::new(256, 1);
+
+        txs.iter()
```

Comment:
Perhaps for a later PR: this step can be parallelized (as noted in the spec). An easy way to achieve this might be with [the rayon crate](https://rust-lang-nursery.github.io/rust-cookbook/concurrency/parallel.html).

Reply:
Thanks for linking `rayon` — this looks great! Updated to add rayon and process in parallel